ups/meas_algorithms.table

```diff
@@ -10,6 +10,7 @@ setupRequired(pipe_base)
 setupRequired(sconsUtils)
 setupRequired(utils)
 setupRequired(cpputils)
+setupRequired(ip_isr)
```

Comment:
This is a bit of a red flag because ip_isr depends on meas_algorithms and so we now get a circular dependency.

Reply:
Correct, and Jenkins failed for the same reason. Is there a better way to do this? My first Jenkins run without this was complaining that it could not import `lsst.ip.isr`.

Reply:
If your test needs ip_isr it means that the test code is probably going to have to go in ip_isr. I haven't looked in detail into what you are doing so I can't really comment further.

Reply:
@arunkannawadi suggested that I copy some bits of code from `ip_isr` explicitly for now

Reply:
If the purpose of the test is to check consistency with `ip_isr`, the test needs to go in `ip_isr` or package that has that and `meas_algorithms` as a dependency, like `pipe_tasks`.  If the code in ip_isr just happens to be convenient for this test, just copy it.

The real test is whether it'd be a problem if you were to copy it and then the `ip_isr` behavior changed to meet ISR's needs without this copy being changed; if it's not a problem, copy away.

Reply:
Besides the isrMock, I think several things from `lsst/ip/isr/isrFunctions.py` should be moved into afw, meas_base, or meas_algorithms. In particular `transposeMaskedImage`, `createPsf`, `growMasks`, and `makeThresholdMask` all seem like they should live in a more generally-accessible place.

That said, the fact that this depends on `ip.isr.Defects` suggests that the Task itself should live in ip_isr or pipe_tasks, not meas_algorithms.

Reply:
@parejkoj's  last point is now moot: we've removed the dependency of the production code on `ip.isr.Defects`.

I agree that at least some of the things he's mentioned should probably be moved to afw or meas_algorithms; the fact that you've found a few ip_isr routines useful for testing this code does suggest that those in particular should be moved up (probably to meas_algorithms).

But moving code like that requires an RFC, so I think we should leave the test in pipe_tasks for now, and then:
 - open an RFC to move some of that general-purpose ip_isr code to meas_algorithms, with a deprecation period for removing aliases to them from ip_isr;
 - once that RFC is implemented, move the unit test for this ticket to meas_algorithms to keep it closer to the code it tests.

That should be a pretty simple RFC and hence good practice for an important procedure.